lib/core/Observer.ts

```diff
@@ -104,13 +111,26 @@ export default class Observer {
 
         // Queue parallel downloading and processing of batch files.
         for (const transaction of transactions) {
-          const awaitingTransaction = {
-            transaction: transaction,
-            processingStatus: TransactionProcessingStatus.Pending
-          };
-          this.transactionsUnderProcessing.push(awaitingTransaction);
-          // Intentionally not awaiting on downloading and processing each operation batch.
-          void this.processTransaction(transaction, awaitingTransaction);
+          if (transaction.transactionTime === this.currentBlockHeight) {
```

Comment:
Another design thought: How about we make another class which does the 'filter-by-rate-limiting' outside of the observer. Then the code in this class changes to (I didn't put too much thought on the naming):
````
constructor (...) {
  ...
  this.operationRateLimitManager = new OperationRateLimitManager(this.versionManager);
  ...
}

private async processTransactions(...) {
  ...
  // This is the new line that rate-limit the existing transactions and the rest of the code remains the same.
  qualifiedTransactions = this.operationRateLimitManager.selectQualifiedTransactions(transactions);
  
  for (const transaction of qualifiedTransactions) {
  ...
}
````

Pros:
1. The new variables + functionality introduced here moves to that new class. 
1. Existing tests of this class don't need to change too much. They just need to spy on this new member ````this.operationLimitManager````.
1. Testing the new class becomes easier as creating test data will be easier.

Reply:
Good idea.